### PR TITLE
ffmpeg: conan v2

### PR DESCRIPTION
**Pull request by paulharris**
Continuing from #15035, pushing here to share the work, feel free to pull the changes into any other PR and go from there.

It currently fails at the configure step, on Windows (DOS environment, compiler=msvc).

Comparing config.log between the old working recipe (with compiler=VisualStudio) and this one,
the difference is the old config.log had:
```
LDFLAGS='-LIBPATH:/k/env/conandata/zlib/1.2.13/cci/13c96f538b52e1600c40b88994de240f/package/fab8ac12fb25ca3eb6ea9dfae033136194574dc3/lib -LIBPATH:/k/env/conandata/bzip2/1.0.8/cci/464be69744fa6d48ed01928cfe470008/package/6d7b0e4f4bdcc313f1817db496d10b1452ad6001/lib .... etc etc
```
whereas the new config.log had
```
LDFLAGS=
```

The new config.log is also missing the lines
```
Platform=x64
UniversalCRTSdkDir='C:\Program Files (x86)\Windows Kits\10\'
VSCMD_ARG_VCVARS_VER=14.3
VSCMD_START_DIR='K:\env\conandata\ffmpeg\5.0\paul\test2\build\a52938edbd2b0d85c72768924adc41517df507a6\build-release'

```
There were a bunch of others, but the main thing firstly is the LDFLAGS, because without -libpath, nothing links.

### CODE REVIEW

**Comment by paulharris** [comment]
@SpaceIm / @prince-chrismc ... I'd like to understand why this keeps failing for me on windows, with errors like:

```
BEGIN ./ffconf.xnuIbwqy/test.c
    1	int main(void){ return 0; }
END ./ffconf.xnuIbwqy/test.c
cl.exe /I/k/env/conandata/vorbis/1.3.7/cci/4bc83814e0814e1a03aa451411f82bd3/package/b1ff2a2f3592c4e70f91c2581c6e0e5ff62d43d4/include /I/k/env/conandata/ogg/1.3.5/cci/1fd1211e20186ca5ff5ba927fb0989ee/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/libwebp/1.2.4/cci/0a7a0849e7780712c650e84e0b0b01c8/package/9099292b4b932fa39f9e092d134fd37ad7ade233/include /I/k/env/conandata/libfdk_aac/2.0.2/cci/c70daa/package/8b3117c31942877c9edf009bf38f3a3b5586c429/include /I/k/env/conandata/libmp3lame/3.100/cci/44b12d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/libvpx/1.11.0/mxrap/2023-02-01/package/5e0af5f73300ab516ed9a8e7ff76a20877f5d36f/include /I/k/env/conandata/libx265/3.4/cci/7347071b88c04ecd3bf66f9d450b6aac/package/d138127af1801535920354a588799ecf562204d9/include /I/k/env/conandata/libx264/cci.20220602/cci/35c2fd77371c19fd564351db10ae6747/package/482564dc3f80b8672f82b4f1806cda8d431a398c/include /I/k/env/conandata/opus/1.3.1/cci/6938146de9da3b6817c85b6d522e8e4d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/opus/1.3.1/cci/6938146de9da3b6817c85b6d522e8e4d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include/opus /I/k/env/conandata/openh264/2.3.1/cci/21db3aa3d786b3e6eb9489dfb88372cb/package/8b3117c31942877c9edf009bf38f3a3b5586c429/include /I/k/env/conandata/openjpeg/2.5.0/cci/cb3ca9c879e991ad313caa16d7e04169/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/openjpeg/2.5.0/cci/cb3ca9c879e991ad313caa16d7e04169/package/b448e8eba5514d8e2baa777bd322fd9583200330/include/openjpeg-2.5 /I/k/env/conandata/libiconv/1.17/cci/7f5704761b99aeddacadf5c7904ffbaa/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/xz_utils/5.4.0/cci/394283edeac4658356894312aa0fdcbb/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/bzip2/1.0.8/cci/464be69744fa6d48ed01928cfe470008/package/8aaea3355c2fe5c6018a0a55628de77a534b1d51/include /I/k/env/conandata/zlib/1.2.13/cci/13c96f538b52e1600c40b88994de240f/package/b448e8eba5514d8e2baa777bd322fd9583200330/include -DOPJ_STATIC -DLZMA_API_STATIC -DNDEBUG -MD -O2 -Ob2 -nologo -c -Fo./ffconf.xnuIbwqy/test.o ./ffconf.xnuIbwqy/test.c
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/vorbis/1.3.7/cci/4bc83814e0814e1a03aa451411f82bd3/package/b1ff2a2f3592c4e70f91c2581c6e0e5ff62d43d4/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/vorbis/1.3.7/cci/4bc83814e0814e1a03aa451411f82bd3/package/b1ff2a2f3592c4e70f91c2581c6e0e5ff62d43d4/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/ogg/1.3.5/cci/1fd1211e20186ca5ff5ba927fb0989ee/package/b448e8eba5514d8e2baa777bd322fd9583200330/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/ogg/1.3.5/cci/1fd1211e20186ca5ff5ba927fb0989ee/package/b448e8eba5514d8e2baa777bd322fd9583200330/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/libwebp/1.2.4/cci/0a7a0849e7780712c650e84e0b0b01c8/package/9099292b4b932fa39f9e092d134fd37ad7ade233/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/libwebp/1.2.4/cci/0a7a0849e7780712c650e84e0b0b01c8/package/9099292b4b932fa39f9e092d134fd37ad7ade233/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/libfdk_aac/2.0.2/cci/c70daa/package/8b3117c31942877c9edf009bf38f3a3b5586c429/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/libfdk_aac/2.0.2/cci/c70daa/package/8b3117c31942877c9edf009bf38f3a3b5586c429/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/libmp3lame/3.100/cci/44b12d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/libmp3lame/3.100/cci/44b12d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/libvpx/1.11.0/mxrap/2023-02-01/package/5e0af5f73300ab516ed9a8e7ff76a20877f5d36f/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/libvpx/1.11.0/mxrap/2023-02-01/package/5e0af5f73300ab516ed9a8e7ff76a20877f5d36f/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/libx265/3.4/cci/7347071b88c04ecd3bf66f9d450b6aac/package/d138127af1801535920354a588799ecf562204d9/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/libx265/3.4/cci/7347071b88c04ecd3bf66f9d450b6aac/package/d138127af1801535920354a588799ecf562204d9/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/libx264/cci.20220602/cci/35c2fd77371c19fd564351db10ae6747/package/482564dc3f80b8672f82b4f1806cda8d431a398c/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/libx264/cci.20220602/cci/35c2fd77371c19fd564351db10ae6747/package/482564dc3f80b8672f82b4f1806cda8d431a398c/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/opus/1.3.1/cci/6938146de9da3b6817c85b6d522e8e4d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/opus/1.3.1/cci/6938146de9da3b6817c85b6d522e8e4d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/opus/1.3.1/cci/6938146de9da3b6817c85b6d522e8e4d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include/opus', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/opus/1.3.1/cci/6938146de9da3b6817c85b6d522e8e4d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include/opus' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/openh264/2.3.1/cci/21db3aa3d786b3e6eb9489dfb88372cb/package/8b3117c31942877c9edf009bf38f3a3b5586c429/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/openh264/2.3.1/cci/21db3aa3d786b3e6eb9489dfb88372cb/package/8b3117c31942877c9edf009bf38f3a3b5586c429/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/openjpeg/2.5.0/cci/cb3ca9c879e991ad313caa16d7e04169/package/b448e8eba5514d8e2baa777bd322fd9583200330/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/openjpeg/2.5.0/cci/cb3ca9c879e991ad313caa16d7e04169/package/b448e8eba5514d8e2baa777bd322fd9583200330/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/openjpeg/2.5.0/cci/cb3ca9c879e991ad313caa16d7e04169/package/b448e8eba5514d8e2baa777bd322fd9583200330/include/openjpeg-2.5', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/openjpeg/2.5.0/cci/cb3ca9c879e991ad313caa16d7e04169/package/b448e8eba5514d8e2baa777bd322fd9583200330/include/openjpeg-2.5' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/libiconv/1.17/cci/7f5704761b99aeddacadf5c7904ffbaa/package/b448e8eba5514d8e2baa777bd322fd9583200330/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/libiconv/1.17/cci/7f5704761b99aeddacadf5c7904ffbaa/package/b448e8eba5514d8e2baa777bd322fd9583200330/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/xz_utils/5.4.0/cci/394283edeac4658356894312aa0fdcbb/package/b448e8eba5514d8e2baa777bd322fd9583200330/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/xz_utils/5.4.0/cci/394283edeac4658356894312aa0fdcbb/package/b448e8eba5514d8e2baa777bd322fd9583200330/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/bzip2/1.0.8/cci/464be69744fa6d48ed01928cfe470008/package/8aaea3355c2fe5c6018a0a55628de77a534b1d51/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/bzip2/1.0.8/cci/464be69744fa6d48ed01928cfe470008/package/8aaea3355c2fe5c6018a0a55628de77a534b1d51/include' ignored
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/zlib/1.2.13/cci/13c96f538b52e1600c40b88994de240f/package/b448e8eba5514d8e2baa777bd322fd9583200330/include', object file assumed
cl : Command line warning D9027 : source file 'I:/k/env/conandata/zlib/1.2.13/cci/13c96f538b52e1600c40b88994de240f/package/b448e8eba5514d8e2baa777bd322fd9583200330/include' ignored
test.c
/k/env/conandata/ffmpeg/5.0/paul/test2/build/a52938edbd2b0d85c72768924adc41517df507a6/src/compat/windows/mslink /LIBPATH:/k/env/conandata/vorbis/1.3.7/cci/4bc83814e0814e1a03aa451411f82bd3/package/b1ff2a2f3592c4e70f91c2581c6e0e5ff62d43d4/lib /LIBPATH:/k/env/conandata/ogg/1.3.5/cci/1fd1211e20186ca5ff5ba927fb0989ee/package/b448e8eba5514d8e2baa777bd322fd9583200330/lib /LIBPATH:/k/env/conandata/libwebp/1.2.4/cci/0a7a0849e7780712c650e84e0b0b01c8/package/9099292b4b932fa39f9e092d134fd37ad7ade233/lib /LIBPATH:/k/env/conandata/libfdk_aac/2.0.2/cci/c70daa/package/8b3117c31942877c9edf009bf38f3a3b5586c429/lib /LIBPATH:/k/env/conandata/libmp3lame/3.100/cci/44b12d/package/b448e8eba5514d8e2baa777bd322fd9583200330/lib /LIBPATH:/k/env/conandata/libvpx/1.11.0/mxrap/2023-02-01/package/5e0af5f73300ab516ed9a8e7ff76a20877f5d36f/lib /LIBPATH:/k/env/conandata/libx265/3.4/cci/7347071b88c04ecd3bf66f9d450b6aac/package/d138127af1801535920354a588799ecf562204d9/lib /LIBPATH:/k/env/conandata/libx264/cci.20220602/cci/35c2fd77371c19fd564351db10ae6747/package/482564dc3f80b8672f82b4f1806cda8d431a398c/lib /LIBPATH:/k/env/conandata/opus/1.3.1/cci/6938146de9da3b6817c85b6d522e8e4d/package/b448e8eba5514d8e2baa777bd322fd9583200330/lib /LIBPATH:/k/env/conandata/openh264/2.3.1/cci/21db3aa3d786b3e6eb9489dfb88372cb/package/8b3117c31942877c9edf009bf38f3a3b5586c429/lib /LIBPATH:/k/env/conandata/openjpeg/2.5.0/cci/cb3ca9c879e991ad313caa16d7e04169/package/b448e8eba5514d8e2baa777bd322fd9583200330/lib /LIBPATH:/k/env/conandata/libiconv/1.17/cci/7f5704761b99aeddacadf5c7904ffbaa/package/b448e8eba5514d8e2baa777bd322fd9583200330/lib /LIBPATH:/k/env/conandata/xz_utils/5.4.0/cci/394283edeac4658356894312aa0fdcbb/package/b448e8eba5514d8e2baa777bd322fd9583200330/lib /LIBPATH:/k/env/conandata/bzip2/1.0.8/cci/464be69744fa6d48ed01928cfe470008/package/8aaea3355c2fe5c6018a0a55628de77a534b1d51/lib /LIBPATH:/k/env/conandata/zlib/1.2.13/cci/13c96f538b52e1600c40b88994de240f/package/b448e8eba5514d8e2baa777bd322fd9583200330/lib -nologo -out:./ffconf.xnuIbwqy/test.exe ./ffconf.xnuIbwqy/test.o
LINK : fatal error LNK1104: cannot open file 'K:\.conan\028900\1\bin\msys64\LIBPATH;K:\env\conandata\vorbis\1.3.7\cci\4bc83814e0814e1a03aa451411f82bd3\package\b1ff2a2f3592c4e70f91c2581c6e0e5ff62d43d4\lib.obj'
C compiler test failed.
```

**Comment by prince-chrismc** [comment]
@jcar87 The GNU triplet is failing in 1.x and needs your magic touch

```
$ conan create all --version 5.1
Exporting the recipe
ERROR: Error loading conanfile at '/Users/christopherm/conan-center-index/recipes/ffmpeg/all/conanfile.py': Unable to load conanfile in /Users/christopherm/conan-center-index/recipes/ffmpeg/all/conanfile.py
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/christopherm/conan-center-index/recipes/ffmpeg/all/conanfile.py", line 13, in <module>
    from conans.tools import get_gnu_triplet
ModuleNotFoundError: No module named 'conans.tools'

$ conan --version
Conan version 2.0.0-beta9
```

**Comment by prince-chrismc** [comment]
I suspect the tools a messing up the paths between the `I:` and `K:` it's doing the wrong things 

```
cl.exe /I/k/env/conandata/vorbis/1.3.7/cci/4bc83814e0814e1a03aa451411f82bd3/package/b1ff2a2f3592c4e70f91c2581c6e0e5ff62d43d4/include /I/k/env/conandata/ogg/1.3.5/cci/1fd1211e20186ca5ff5ba927fb0989ee/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/libwebp/1.2.4/cci/0a7a0849e7780712c650e84e0b0b01c8/package/9099292b4b932fa39f9e092d134fd37ad7ade233/include /I/k/env/conandata/libfdk_aac/2.0.2/cci/c70daa/package/8b3117c31942877c9edf009bf38f3a3b5586c429/include /I/k/env/conandata/libmp3lame/3.100/cci/44b12d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/libvpx/1.11.0/mxrap/2023-02-01/package/5e0af5f73300ab516ed9a8e7ff76a20877f5d36f/include /I/k/env/conandata/libx265/3.4/cci/7347071b88c04ecd3bf66f9d450b6aac/package/d138127af1801535920354a588799ecf562204d9/include /I/k/env/conandata/libx264/cci.20220602/cci/35c2fd77371c19fd564351db10ae6747/package/482564dc3f80b8672f82b4f1806cda8d431a398c/include /I/k/env/conandata/opus/1.3.1/cci/6938146de9da3b6817c85b6d522e8e4d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/opus/1.3.1/cci/6938146de9da3b6817c85b6d522e8e4d/package/b448e8eba5514d8e2baa777bd322fd9583200330/include/opus /I/k/env/conandata/openh264/2.3.1/cci/21db3aa3d786b3e6eb9489dfb88372cb/package/8b3117c31942877c9edf009bf38f3a3b5586c429/include /I/k/env/conandata/openjpeg/2.5.0/cci/cb3ca9c879e991ad313caa16d7e04169/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/openjpeg/2.5.0/cci/cb3ca9c879e991ad313caa16d7e04169/package/b448e8eba5514d8e2baa777bd322fd9583200330/include/openjpeg-2.5 /I/k/env/conandata/libiconv/1.17/cci/7f5704761b99aeddacadf5c7904ffbaa/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/xz_utils/5.4.0/cci/394283edeac4658356894312aa0fdcbb/package/b448e8eba5514d8e2baa777bd322fd9583200330/include /I/k/env/conandata/bzip2/1.0.8/cci/464be69744fa6d48ed01928cfe470008/package/8aaea3355c2fe5c6018a0a55628de77a534b1d51/include /I/k/env/conandata/zlib/1.2.13/cci/13c96f538b52e1600c40b88994de240f/package/b448e8eba5514d8e2baa777bd322fd9583200330/include -DOPJ_STATIC -DLZMA_API_STATIC -DNDEBUG -MD -O2 -Ob2 -nologo -c -Fo./ffconf.xnuIbwqy/test.o ./ffconf.xnuIbwqy/test.c
cl : Command line warning D9024 : unrecognized source file type 'I:/k/env/conandata/vorbis/1.3.7/cci/4bc83814e0814e1a03aa451411f82bd3/package/b1ff2a2f3592c4e70f91c2581c6e0e5ff62d43d4/include', object file assumed
```

File: recipes/ffmpeg/all/conanfile.py [hunk 1]
```diff
@@ -521,71 +527,97 @@ def opt_append_disable_if_set(args, what, v):
...
+        args2["--includedir"] = None
+        args2["--oldincludedir"] = None
+
+        tc.configure_args.extend(args)
+        tc.update_configure_args(args2)
+
+        tc.generate()
+
+
+        if is_msvc(self):
```
Comment (prince-chrismc) [inline recipes/ffmpeg/all/conanfile.py:601]: I stole this from `recipes/coin-utils/all/conanfile.py` maybe see if you can build that one?


```
        env = tc.environment()
        if is_msvc(self):
            compile_wrapper = unix_path(self, self.conf.get("user.automake:compile-wrapper", check_type=str))
            ar_wrapper = unix_path(self, self.conf.get("user.automake:lib-wrapper", check_type=str))
            env.define("CC", f"{compile_wrapper} cl -nologo")
            env.define("CXX", f"{compile_wrapper} cl -nologo")
            env.define("LD", "link -nologo")
            env.define("AR", f"{ar_wrapper} \"lib -nologo\"")
            env.define("NM", "dumpbin -symbols")
            env.define("OBJDUMP", ":")
            env.define("RANLIB", ":")
            env.define("STRIP", ":")
        tc.generate(env)
```

File: recipes/ffmpeg/all/conanfile.py [hunk 2]
```diff
@@ -521,71 +527,97 @@ def opt_append_disable_if_set(args, what, v):
...
+            apple_min_version_flag = AutotoolsToolchain(self).apple_min_version_flag
+            if apple_min_version_flag:
+                extra_cflags.append(apple_min_version_flag)
+                extra_ldflags.append(apple_min_version_flag)
+        if is_msvc(self):
+            pkg_config = os.getenv("PKG_CONFIG") # TODO
+            args.append(f"--pkg-config={pkg_config}")
             args.append("--toolchain=msvc")
-            if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) <= "12":
+            if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) <= "12":
```
Comment (prince-chrismc) [inline recipes/ffmpeg/all/conanfile.py:564]: in conan 1.57 they added check min vs with raise = false to be 2.0 compatible

**Pull request closed without merging**